galite-core/src/main/kotlin/org/kopi/galite/visual/VItemTree.kt

```diff
@@ -15,6 +15,285 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
+
 package org.kopi.galite.visual
 
-class VItemTree 
\ No newline at end of file
+import java.awt.event.KeyEvent
+import java.util.Locale
+
+import javax.swing.tree.TreeNode
+
+import kotlin.collections.ArrayList
+import kotlin.collections.List
+import kotlin.collections.indices
+import kotlin.system.exitProcess
+
+import org.kopi.galite.l10n.LocalizationManager
+import org.kopi.galite.util.base.InconsistencyException
+
+/**
+ * Presents a new instance of VItemTree.
+ * @param rootName the rot item name
+ * @param items the items list
+ * @param isInsertMode enable Add and Remove items
+ * @param selectionType item selection mode :
+ * 1 : no edit
+ * 2 : Single selection
+ * 3 : Multi selection
+ * 4 : Multi selection with default value
+ * @param isLocalised if true, enable item localisation
+ * @param itemTreeManager the tree save manager
+ * @param isRemoveDescendantsAllowed if true, remove item descendants when removig item
+ * @param nameLength max length of item name
+ * @param localisedNameLength max length of item localisation
+ * @param descriptionLength max length of item description
+ */
+class VItemTree(rootName: String?,
```

Comment:
why nullable ?

Reply:
we have this in our code :  
`private val rootName: String = rootName ?: "Items"`